language/stdlib/modules/libra.move

```diff
@@ -1,23 +1,50 @@
 address 0x0:
 
 module Libra {
+    use 0x0::Association;
     use 0x0::Transaction;
     use 0x0::Vector;
+    use 0x0::FixedPoint32;
+    use 0x0::RegisteredCurrencies;
+    use 0x0::LibraConfig;
 
-    // A resource representing a fungible token
-    resource struct T<Token> {
-        // The value of the token. May be zero
-        value: u64,
-    }
+    // The currency has a `CoinType` color that tells us what currency the
+    // `value` inside represents.
+    resource struct T<CoinType> { value: u64 }
 
-    // A singleton resource that grants access to `Libra::mint`. Only the Association has one.
-    resource struct MintCapability<Token> { }
+    // A minting capability allows coins of all types to be minted.
+    resource struct MintCapability<CoinType> { }
+
+    // A operations capbility to allow this module to register currencies
+    // with the RegisteredCurrencies on-chain config.
+    resource struct CurrencyRegistrationCapability {
+        cap: RegisteredCurrencies::RegistrationCapability,
+    }
 
-    resource struct Info<Token> {
-        // The sum of the values of all Libra::T resources in the system
+    // The information for every supported currency is stored in a resource
+    // under the `currency_addr()` address. Unless they are specified
+    // otherwise the fields in this resource are immutable.
+    resource struct CurrencyInfo<CoinType> {
+        // The total value for the currency represented by
+        // `CoinType`. Mutable.
         total_value: u128,
         // Value of funds that are in the process of being burned
         preburn_value: u64,
+        // The (rough) exchange rate from `CoinType` to LBR.
+        to_lbr_exchange_rate: FixedPoint32::T,
+        // Holds whether or not this currency is synthetic (contributes to the
+        // off-chain reserve) or not. An example of such a currency would be
+        // the LBR.
+        is_synthetic: bool,
+        // e.g. 10^6 for the Coin1
+        scaling_factor: u64,
+        // The smallest fractional part for the currency (e.g. 10^2 for cents)
```

Comment:
Maybe say explicitly what the unit is + connect it to the example? I think this is saying `fractional_part` would be 2 for LibraUSD with two decimal places (hence 10^2), but not sure.